ROADMAP.md

```diff
@@ -0,0 +1,82 @@
+
+*see*: [#2461](https://github.com/grafana/k6/issues/2461)
+
+### Javascript ES Modules support
+
+Although k6 supports modules, we currently rely on Babel for code transpilation, which has performance and design drawbacks. We're working on implementing native ES module support in Goja, our JavaScript runtime. The project is ongoing, with no estimated delivery date.
+
+*see*: [#2258](https://github.com/grafana/k6/issues/2258)
+
+### Large files handling
+
+We're working to improve k6's handling of large files, ensuring efficient and reliable test execution for users working with big data sets or complex test scenarios. Our current file-handling approach can lead to significant memory consumption in some cases. We're addressing this issue, and specific improvements will be incrementally released in upcoming versions.
+
+*see*: [#2974](https://github.com/grafana/k6/issues/2974)
+
+### Improve `async/await` syntax support
+
+We aim to enhance async function and operation support within k6, making it easier for users to write and execute asynchronous code. We're improving compatibility with our existing APIs and expect users to see improvements in upcoming versions.
+
+*see*: [#3014](https://github.com/grafana/k6/issues/3014), [#2967](https://github.com/grafana/k6/issues/2967), [#2869](https://github.com/grafana/k6/issues/2869)
+
+### Command-line output improvements
+
+We've received consistent feedback about potential improvements to the k6 terminal summary displayed at the end of a test run. We aim to enhance the CLI output to provide better readability and more insightful information, helping users understand their test results and performance more efficiently.
+
+## Long-term goals
+
+These are goals we anticipate will take **a year or longer to achieve**. We have yet to set a specific timeline for these goals.
+
+### Native distributed execution
+
+We aim to enhance k6 to natively support distributed load testing across multiple machines, making it easier for users to scale their tests. Although some research and initial design work have been done, we're still in the early stages of development and still need an estimated delivery date.
+
+*see*: [#140](https://github.com/grafana/k6/issues/140)
+
+### Test suites
+
+We aim to implement a test suite feature to enable better organization and management of test cases, making it more convenient for users to maintain and execute multiple tests. The end goal from a user perspective would be to be able to run multiple scripts, one after the other, in a user-defined sequence to form what we refer to as a "test suite". This would allow, for instance, to have scenarios wait for other scenarios to complete before they start their execution.
+
+*see*: [#1342](https://github.com/grafana/k6/issues/1342)
+
+### HDR & sparse histograms
```

Comment:
Both test suites and HDR histograms seem like mid-term goals, no? We have some PoCs and research on both, and they're relatively minor features compared to some of the other larger projects.

Or maybe we should change the definition of long-term goals? Instead of "These are goals we anticipate will take a year or longer to achieve", maybe "these are goals that are planned to be delivered more than a year from now". I.e. not how long they will take to achieve, but when we plan to deliver them.

Reply:
Super good point indeed on the phrasing, thanks a lot 🙇🏻  Will update accordingly 👍🏻